src/main/java/su/litvak/chromecast/api/v2/Channel.java

```diff
@@ -424,8 +441,13 @@ public Status launch(String appId) throws IOException {
     }
 
     public Status stop(String sessionId) throws IOException {
-        StandardResponse.Status status = sendStandard("urn:x-cast:com.google.cast.receiver", StandardRequest.stop(sessionId), DEFAULT_RECEIVER_ID);
-        return status == null ? null : status.status;
+        stopRequested = true;
+        try {
+            StandardResponse.Status status = sendStandard("urn:x-cast:com.google.cast.receiver", StandardRequest.stop(sessionId), DEFAULT_RECEIVER_ID);
+            return status == null ? null : status.status;
+        } finally {
+            stopRequested = false;
```

Comment:
I think you are right and my bad that I have actually missed the fact that request is posted synchronously.